dot/types/block.go

```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
+// BlockNumber represents a block number
+type BlockNumber uint32
```

Comment:
Block numbers are actually type uint in our code, could this cause possible issues for encoding/decoding? 

Following up on this, what is the reason for this type being created? Is it just for readability? I'm not opposed to it, just want to make sure I understand its context

Reply:
> Block numbers are actually type uint in our code, could this cause possible issues for encoding/decoding?

substrate uses `u32`. What's the reason for using `uint`?

> what is the reason for this type being created? Is it just for readability?

Yes, and usability

Reply:
> Block numbers are actually type uint in our code

No, BlockNumbers are uint32 at most places in gossamer. `types.Header` uses uint though. `BlockAnnounceHandshake`, `ProveFinalityRequest` and many other places use uint32. I do remember a discussion about this and I think substrate+polkadot itself doesn't use the same type for block number across their codebase.

In this particular situation, we don't have a choice to choose between uint and uint32. We have to make our types identical to types used by polkadot. So, we have to use uint32.

>  could this cause possible issues for encoding/decoding?

If I rememeber well, uint and uint32 would produce different encoding.